cpp/src/io/parquet/page_delta_decode.cu

```diff
@@ -559,43 +559,41 @@ CUDF_KERNEL void __launch_bounds__(decode_block_size)
     uint32_t target_pos;
     uint32_t const src_pos = s->src_pos;
 
-    if (t < 3 * warp_size) {  // warp 0..2
+    if (warp.meta_group_rank() < 3) {  // warp 0..2
       target_pos = min(src_pos + 2 * batch_size, s->nz_count + s->first_row + batch_size);
     } else {  // warp 3
       target_pos = min(s->nz_count, src_pos + batch_size);
     }
     // this needs to be here to prevent warp 3 modifying src_pos before all threads have read it
-    __syncthreads();
+    block.sync();
 
     // warp0 will decode the rep/def levels, warp1 will unpack a mini-batch of prefixes, warp 2 will
     // unpack a mini-batch of suffixes. warp3 waits one cycle for warps 0-2 to produce a batch, and
     // then stuffs values into the proper location in the output.
-    if (t < warp_size) {
+    if (warp.meta_group_rank() == 0) {
       // decode repetition and definition levels.
       // - update validity vectors
       // - updates offsets (for nested columns)
       // - produces non-NULL value indices in s->nz_idx for subsequent decoding
-      gpuDecodeLevels<delta_rolling_buf_size, level_t>(s, sb, target_pos, rep, def, t);
-
-    } else if (t < 2 * warp_size) {
+      gpuDecodeLevels<delta_rolling_buf_size, level_t>(s, sb, target_pos, rep, def, warp);
+    } else if (warp.meta_group_rank() == 1) {
       // warp 1
```

Comment:
we might not need these comments now, the condition is pretty self-explanatory

Reply:
I think it's ok to still keep them for readability